python/lib/evaluations/dmod/evaluations/evaluate.py

```diff
@@ -529,9 +533,12 @@ def score(
 
         scores: typing.Dict[typing.Tuple[str, str], metrics.MetricResults] = dict()
 
+        # TODO: Distribute each group to different processes - utilize dmod.core.common.collections.AccessCache
```

Comment:
@christophertubbs, I don't quite understand the larger design you have in mind enough to see how this caching construct will make it easier to distribute data and perform the evaluations across multiple processes.  E.g., why not just load chunks and stick those in a [multiprocess.Queue](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Queue)?  

Before finishing my review, can you walk/talk me through things a little more?  Let me know if you think we should do that synchronously.

Reply:
Complex collections of values are getting passed around and any sort of possible mutation wouldn't necessarily get passed along correctly when bundled up into large objects. This here is not a particularly grand example of where anything might get mutated. The cache is to be used for a lot of the io work for the backend classes. The required data can end up being fairly varied, so you might end up with a lot of artifacts for thresholds, very few, none at all, same with location data, observations, forecasts, etc. To handle that, we'd end up with multiple queues with weird polling behavior due many processes attempting to access its contents like a key-value store by sequentially popping and pushing, all at the same time.

The structure is also missing some of what needs to be available for proper multiprocessing work. A PR will be coming down the line that will have _those_ implementations.

Maybe we can do the walk/talk next week. This here is just a tiny piece of the overall puzzle, so it's hard to show how things are going to be wired up.